src/idl_gen_python.cpp

```diff
@@ -402,21 +428,43 @@ class PythonGenerator : public BaseGenerator {
   // Recursively generate struct construction statements and instert manual
   // padding.
   void StructBuilderBody(const StructDef &struct_def,
-                         const char *nameprefix, std::string *code_ptr) {
+                         const char *nameprefix, std::string *code_ptr, size_t index = 0, bool in_array = false) {
     std::string &code = *code_ptr;
-    code += "    builder.Prep(" + NumToString(struct_def.minalign) + ", ";
+    std::string indent(index * 4, ' ');
+    code += indent + "    builder.Prep(" + NumToString(struct_def.minalign) + ", ";
     code += NumToString(struct_def.bytesize) + ")\n";
     for (auto it = struct_def.fields.vec.rbegin();
         it != struct_def.fields.vec.rend(); ++it) {
       auto &field = **it;
+      auto type = field.value.type;
       if (field.padding)
-        code += "    builder.Pad(" + NumToString(field.padding) + ")\n";
+        code += indent + "    builder.Pad(" + NumToString(field.padding) + ")\n";
       if (IsStruct(field.value.type)) {
         StructBuilderBody(*field.value.type.struct_def,
-                          (nameprefix + (NormalizedName(field) + "_")).c_str(), code_ptr);
+                          (nameprefix + (NormalizedName(field) + "_")).c_str(),
+                          code_ptr, index, in_array);
       } else {
-        code += "    builder.Prepend" + GenMethod(field) + "(";
-        code += nameprefix + MakeCamel(NormalizedName(field), false) + ")\n";
+        auto index_var = "_idx" + NumToString(index);
+        if (IsArray(field.value.type)) {
+          code += indent + "    for " + index_var + " in range(";
+          code += NumToString(field.value.type.fixed_length);
+          code += " , 0, -1):\n";
+          type = type.VectorType();
+          in_array = true;
+        }
+        if (IsStruct(type)) {
+          StructBuilderBody(*field.value.type.struct_def,
+                            (nameprefix + (NormalizedName(field) + "_")).c_str(),
+                            code_ptr, index + 1, in_array);
+        } else {
+          code += IsArray(field.value.type) ? "    " : "";
+          code += indent + "    builder.Prepend" + GenMethod(field) + "(";
+          code += nameprefix + MakeCamel(NormalizedName(field), false);
+          for (size_t i = 0; in_array && i < index + IsArray(field.value.type) ? 1 : 0; i++) {
```

Comment:
Please, set priority in the condition checking with `()` brackets.
Or remove `IsArray(field.value.type) ? 1 : 0` out of loop.

PVS-Studio high-pri warning:
> V502 Perhaps the '?:' operator works in a different way than it was expected. The '?:' operator has a lower priority than the '+' operator. idl_gen_python.cpp 463

https://www.viva64.com/en/w/v502/print/

Reply:
Good catch!

Reply:
You can use `!!IsArray(field.value.type)` instead of `(IsArray(field.value.type) ? 1 : 0)`